### DISCUSSION TITLE
java集成iotdb集群

### DISCUSSION THREAD

**Discussion by kouki188** [apache/iotdb#12054] [Q&A]
在使用java集成iotdb集群模式时，demo是单机模式下相同的，也跑通了的，主要修改的地方是session = new Session(nodeList,user, password,fetchSize); 这里是把三个节点的url填写进去的，但是遇到了问题，查询是成功的，插入一直失败，但是没有任何的报错；我在linux后台使用脚本的插入、查询、同步都是没问题的，排除了部署的问题之后，不知道java代码还需要如何修改？

**Comment by HTHou** [comment]
可以提供一下写入程序代码？

**Reply by kouki188** [reply]
解决了，下了官网的代码，可以跑通了，好像是因为version(Version.V_1_0)，没设置这个的问题；但是在执行了插入代码后，会提示无法重定向到其他节点的报错，但是其他节点明明已经同步成功了
![1708413429980](https://github.com/apache/iotdb/assets/160562350/f046931b-2b26-4523-a29c-f19c2991fe08)

**Reply by kouki188** [reply]
好了好了，在本地设置了hosts就可以了
